src/main/java/frc/robot/vision/limelights/MultiLimelights.java

```diff
@@ -26,27 +29,35 @@ public void updateGyroAngles(GyroAngleValues gyroAngleValues) {
 		}
 	}
 
-	public List<LimelightRawData> getAllAvailableLimelightData() {
-		List<LimelightRawData> limelightsData = new ArrayList<>();
+	public List<VisionRawData> getAllAvailableLimelightData() {
+		List<VisionRawData> limelightsData = new ArrayList<>();
 
 		for (Limelight limelight : limelights) {
 			limelight.updateLimelight();
 			Optional<Pair<Pose3d, Double>> observation = limelight.getUpdatedPose3DEstimation();
 
 			if (observation.isPresent()) {
-				LimelightRawData limelightRawData = new LimelightRawData(
+				double timestamp = observation.get().getSecond();
+				VisionRawData visionData = new VisionRawData(
+					limelight.getCameraName(),
 					observation.get().getFirst(),
-					limelight.getAprilTagHeight(),
-					limelight.getDistanceFromAprilTag(),
-					observation.get().getSecond()
+					calculateAmbiguity(limelight.getAprilTagHeight()),
+					timestamp,
+					limelight.getLatency()
 				);
-				limelightsData.add(limelightRawData);
+				limelightsData.add(visionData);
 			}
 		}
 
 		return limelightsData;
 	}
 
+	// we should find a better function later
+	private double calculateAmbiguity(double aprilTagHeightMeters) {
+		double heightPower3 = Math.pow(aprilTagHeightMeters, 3);
+		return heightPower3 / (heightPower3 + 200);
+	}
```

Comment:
we can't calculate ambiguity from distance to apriltag, that's not the same and I don't think they correlate